### DISCUSSION TITLE
Explorador

### DISCUSSION THREAD

**Discussion by GeraldinTorres** [Launch-X-Latam/MisionFrontEnd#274] [Q&A]
Buenas, tengo un problema, a la hora de abrir mi archivo html en el explorador pare ver como va mi progreso me sale en blanco no se actualiza, por mas que haga no se actualiza, volvi a empezar y tengo el mismo problema, no se a que se deba, alguien me podria ayudar.?

**Comment by OscarJG20** [comment]
¿Qué editor de texto o IDE utilizas?

a) En el caso de que sea Visual Studio Code, existe un plugin "Live server" que te permite observar los cambios en tiempo real (ya no necesitas darle clic al HTML como tal).
b) Hay veces que los cambios no se visualizan porque se quedan en el caché de tu navegador, aprieta CTRL + F5, es para forzar la carga de la página completa (manda a llamar todos los archivos).

Esas son las cosas que se me ocurren, espero te ayuden.

**Comment by Cerecero** [comment]
Puedes postear imagenes de tu codigo, navegador y el archivo, para poder ver a mejor detalle el problema.

**Comment by iam-leo** [comment]
Hola Geraldin, chequeaste que tu archivo tenga extensión .html? por ej `index.html`
a otras personas le sucedió eso, no visualizaban nada por tener mal escrito la extensión .html

**Comment by GeraldinTorres** [comment]
![image](https://user-images.githubusercontent.com/114269218/198752209-4062b3f0-beed-4e47-a3b2-ccbc049fa546.png)
Este es el codigo 

![image](https://user-images.githubusercontent.com/114269218/198752286-87300657-1659-45b7-af1d-4832444a4d40.png)
Si lo abro con la extencion se me actualiza, pero si le hago otros cambios y lo quiero refrescar no pasa nada, ademas de que el link no sirve.

**Reply by iam-leo** [reply]
Tienes Live Server instalado en VS Code?

**Comment by Amhedriel** [comment]
# MUY BUENAS
Hola Geraldin, con respecto a tu pregunta, si entendí bien es que no puedes visualizar tu archivo **.html** como quieres en el navegador y compartir la dirección, si es así te explico:

* Si usas VSCode (Visual Studio Code)

![Captura de pantalla (286)](https://user-images.githubusercontent.com/61332909/198753107-0d6b79fc-87c4-4c63-b36c-03236d3a6c7f.png "pantalla de mi escritorio de Visual Studio Code")

* Busca la extensión de Live Server, búscalo en el apartado de extensiones.

![Captura de pantalla (287)](https://user-images.githubusercontent.com/61332909/198753204-6c249c13-8cd1-4adf-8f6c-6e4d83f64db0.png "Extensiones")

* Lo puedes ver así.

![Captura de pantalla (288)](https://user-images.githubusercontent.com/61332909/198753254-082ace32-4502-484e-b90a-720dc3c4f026.png "Pestaña de Live Server")

* Una vez instalado, puede que te pida reiniciar el IDE para aplicarlo, después de eso verás que se te agrego en la parte de abajo de tu pantalla de trabajo que tiene un símbolo de antena y dice **Go Live**

![Captura de pantalla (289)](https://user-images.githubusercontent.com/61332909/198753376-3d1dd9ef-57ea-484d-9049-5f7a152cd603.png "Go Live")

* Entonces se abrirá una página web en tu navegador configurado como el navegador predeterminado (en mi caso EDGE) y te mostrará lo que tienes tipeado en tu IDE, la diferencia radica en que el resultado que te muestra es como se mostrará en los navegadores actuales, con todos los cabios y errores que tenga tu código *HTML*

![Captura de pantalla (290)](https://user-images.githubusercontent.com/61332909/198753469-be6f3a4b-724f-459e-a6f8-8cb17dd3610e.png "Navegador abierto")

* No solo se abrirá tu navegador y ya, lo que hace es simular ser un servidor y mostrarte la página tal como se vería en un ambiente real de producción, a diferencia de lo que nos muestras, que la forma en la que lo abriste es ejecutando tu archivo, por eso tienes como dirección tu PC dígase `C:\User\Pamela%20Torres\...` lo que tienes que hacer es abrirlo mediante el Live Server y te saldrá una dirección que está ejecutando en un servidor.

![IP](https://user-images.githubusercontent.com/61332909/198753642-f78b38f4-2cb0-429b-8f96-fb734e529ce7.png "Dirección de server")

Ahora bien, aclarado ese tema, tienes que guardar tu progreso en VSCode con `Ctrl + S` y cuando vuelvas a la página en la que ya ejecutaste tu Live Server entonces actualízalo y verás los cambios.

**NOTA** 
A pesar de que ya sea un servidor, no significa que puedas copiar esa dirección y copiarla a GitHub sin más, que así no funciona, tienes que subirlo a tu repositor de GitHub con Git y entrando a esa página copia la dirección y podrás compartir tus resultados.

Práctica de Innovacción a medio hacer de =P [html](https://github.com/Amhedriel/PracticasInnovaccion/blob/main/HTML/README.md "Ir a mi repositorio") aún no tengo el esqueleto HTML pero lo terminare este fin de semana y lo subo jeje, pero espero haberte dado una idea de lo que buscas hacer, comparte este hilo si alguien tiene un problema similar y así evitamos repetir todo el proceso XD.

Saludos.

**Comment by GeraldinTorres** [comment]
A ok entiendo, entonces el link que me genera el live server no es un link funcional por asi decirlo.? es por eso que no lo puedo abrir.?

**Reply by Amhedriel** [reply]
No es funcional para enviarlo como link a otras personas o dispositivos (porque para eso se debe hacer una configuración especial, pero se aleja mucho de los conocimientos actualmente adquiridos) sin mencionar que deberías tener tu PC encendido todo el día como un verdadero servidor.

Para eso: sube tus archivos con >Git a > GitHub, luego configura tu GitHub para poder convertirlo en un Host y mostrar tus avances, porque ese link si se podrá compartir.

**Comment by doguedogue** [comment]
Hola, si tu abres un archivo html con un navegador puedes ver esto:
file:///C:/tmp/MI_PAGINA.html  esta es la ruta donde tengo guardado el archivo en mi equipo.
![image](https://user-images.githubusercontent.com/23409026/198843717-f54b58d5-e9ab-4754-9466-25297c6c15cf.png)

Pero cuando haces tus pruebas locales puedes publicarlas en servidores web **locales** como XAMP, Apache, LiveServer, etc.
Estos servidores locales utilizan la tu dirección IP o un alias por ejemplo localhost o 127.0.0.1.
Le adicionan a la url un puerto (ej. 5500) con el que esta configurado el servidor web local en tu equipo ej. de LiveServer
[127.0.0.1:5500](http://127.0.0.1:5500/MI_PAGINA.html) esta es la url LOCAL no la puedes compartir y que alguien mas la vea fuera de tu red.
![image](https://user-images.githubusercontent.com/23409026/198843953-53a0a510-6c3e-4e4c-93b3-65cb7253b7bf.png)

Las páginas web se visualizan o publican o despliegan en servidores web como GitHub (u otros) si subes tu página y la despliegas con GitHub pages tendras una URL pública en internet para compartir.
https://TU_USUARIO_GITHUB.github.io/MI_REPOSITORIO/MI_PAGINA.html   página desplegada con GitHub pages 
![image](https://user-images.githubusercontent.com/23409026/198844154-0f40b53e-49f9-4f82-b91d-bc2c2bec4701.png)

**Comment by GeraldinTorres** [comment]
Aaaaa ya todo tiene sentido jaja, muchas gracias me ayudaron un monton.
